### PR TITLE
Drop ExternalInfraCluster CRD in favour of AWSCluster

### PR DESCRIPTION
**Pull request by enxebre**
This PR drops the ExternalInfraCluster CRD in favour of AWSCluster.

Originally we added support for externally managed infra in CAPA via the ExternalInfraCluster CRD kubernetes-sigs/cluster-api-provider-aws#2124 and we used that commit of CAPA in hypershift.

Later on we decided to revert that approach upstream and reuse the existing ${infra}Cluster CRDs with an annotation to support externally managed infrastructure kubernetes-sigs/cluster-api#4135

This PR bring latest CAPI/CAPA.

As a follow up we need to rebuild the images and consume them from quay.io/hypershift.

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -340,10 +332,24 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to reconcile hostedcontrolplane: %w", err)
 	}
 
+	// Reconcile external AWSCluster
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(hcp), hcp); err != nil {
+		r.Log.Error(err, "failed to get control plane ref")
+		return reconcile.Result{}, err
+	}
+
+	awsCluster := controlplaneoperator.ExternalInfraCluster(controlPlaneNamespace.Name, hcluster.Name)
```
Comment (csrwng) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:341]: should this only happen when Platform.Type == "AWS" ? Also s/ExternalInfraCluster/AWSCluster/
do we still create/reconcile an AWSCluster when Platform.Type == "None"?

Comment (enxebre) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:341]: >should this only happen when Platform.Type == "AWS
yes.
>do we still create/reconcile an AWSCluster when Platform.Type == "None"?
No.

Each provider must have their own infraCluster CR.

Let me update.

**Comment by enxebre** [comment]
Rebased after https://github.com/openshift/hypershift/pull/269
/test e2e-aws

**Comment by enxebre** [comment]
Pushed changes so as discussed we'll run the aws controllers for platform None for now.
https://github.com/openshift/hypershift/pull/272/files#diff-b5cde720842b19b53da3f454d827e27327c4559b2e4849906dac728586f51eb3R356-R358
/test e2e-aws

**Pull request merged**